### PR TITLE
Update actions

### PR DESCRIPTION
**Pull request by r-oung**
- Updates action packages
- Adds builds strategy for Ubuntu 20.04 and macOS 11.0
- Adds builds strategy for QT 5.12.9

### CODE REVIEW

File: .github/workflows/ubuntu_build.yml [hunk 1]
```diff
@@ -5,29 +5,30 @@ on:
...
   build:
-    name: Test build on ${{ matrix.os }} with Qt ${{ matrix.qt_version }}
+    name: Build on ${{ matrix.os }} with Qt ${{ matrix.qt_version }}
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: ['ubuntu-latest']
-        qt_version: ['5.12.6']
+        os: ['ubuntu-18.04', 'ubuntu-20.04']
+        qt_version: ['5.12.6', '5.12.9']
```
Comment (philippReist) [inline .github/workflows/ubuntu_build.yml:21]: Did you test UI functionality with 5.12.9?

**Comment by philippReist** [comment]
Let me do a UI functionality test as well with 5.12.9 / maybe it makes sense to see if it works properly with 5.15 LTS.

**Comment by philippReist** [comment]
Just tested with 5.15.2 - just a few error messages that I think I can fix.

**Comment by r-oung** [comment]
Sorry, I didn't see your response to this PR for some reason.

I originally created this PR in order to support the latest macOS and Ubuntu OS. While I was doing so, I also decided to update the QT action too. I added 5.12.9 because that was the latest version "officially" supported by the [github install-qt-action](https://github.com/jurplel/install-qt-action).

I've tested this on 5.12.9 and 5.12.10. It builds properly, passes all unit tests, and passes functionality test.

I also tried testing this with 5.15.2, but I get the following errors at run-time:
```
qrc:/qml/components/FrequencySlider.qml:21: TypeError: Cannot read property 'primitive' of null
qrc:/qml/components/FrequencySlider.qml:21: TypeError: Cannot read property 'primitive' of null
qrc:/qml/components/TimerBar.qml:51:3: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/components/TimerBar.qml:51:3: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/main.qml:167:5: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/main.qml:162:5: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/components/AudioControl.qml:51:3: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/components/AudioControl.qml:38:3: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/components/LEDPrimitiveControl.qml:47:3: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/components/MotorPrimitiveControl.qml:56:3: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/components/MP3FileControl.qml:60:3: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/components/MP3FileControl.qml:48:2: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/main.qml:72:1: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
qrc:/qml/components/LEDPrimitiveControl.qml:357: TypeError: Cannot call method 'updatePrimitive' of null
qrc:/qml/components/MotorPrimitiveControl.qml:653: TypeError: Cannot call method 'updatePrimitive' of null
qrc:/qml/components/MotorPrimitiveControl.qml:468: TypeError: Cannot read property 'primitive' of null
qrc:/qml/components/MotorPrimitiveControl.qml:208: TypeError: Cannot read property 'primitive' of null
```

By all means, please fix the errors. However, I'm not sure if we should support 5.15.2 in CI/CD right now because the [github install-qt-action](https://github.com/jurplel/install-qt-action) only supports 5.12.9 (according to the README). Your thoughts?

**Review by philippReist — APPROVED** [review]
looks good to me!

**Comment by philippReist** [comment]
Ah good point - yes then let's stay with 5.12.9. I will do a functionality test this week and then I can update master.

**Comment by r-oung** [comment]
@philippReist don't forget to merge this in after testing on 5.12.9.

**Comment by philippReist** [comment]
@r-oung: I rebased the audiochannel swapping MR on top of this so I think we can close this without merging. But best if you double-check that develop has these changes in.

**Comment by r-oung** [comment]
@philippReist looks good to me. Closing this PR.

**Pull request closed without merging**